integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/command/SICreationCommand.scala

```diff
@@ -209,6 +209,11 @@ private[sql] case class CarbonCreateSecondaryIndexCommand(
             .get
         }")
       }
+      val isSpatialColPresent = dims.find(x => x.getColumnSchema.isSpatialColumn)
```

Comment:
Please check , I think we need to remove isSpatialColumn from columnSchema. 
we used that mainly for making  it invisible. Now as it is visible. It is just another plan column. Instead we can check if column name is in the table property or not.

@VenuReddy2103 @ShreelekhyaG

Reply:
removed isSpatialColumn from schema.